staging_vespalib/src/vespa/vespalib/util/singleexecutor.cpp

```diff
@@ -158,11 +158,13 @@ SingleExecutor::wait_for_room(Lock & lock) {
     if (taskLimit != _wantedTaskLimit.load(std::memory_order_relaxed)) {
         drain(lock);
         _tasks = std::make_unique<Task::UP[]>(_wantedTaskLimit);
+        double waterMarkRatio = double(get_watermark()) / _taskLimit.load(std::memory_order_relaxed);
```

Comment:
Consider calculating a watermark ratio in the constructor and keeping the ratio as a member variable.
Repeated task limit adjustments can cause watermark to become lower and lower due to rounding.

Reply:
Thanks, done